applications/luci-app-usteer/htdocs/luci-static/resources/view/usteer/usteer.js

```diff
@@ -144,22 +169,55 @@ function collectWlanAPInfos(compactconnectioninfo_table_entries, wlanAPInfos) {
 	}
 };
 
+var callNetworkRrdnsLookup = rpc.declare({
+	object: 'network.rrdns',
+	method: 'lookup',
+	params: [ 'addrs', 'timeout', 'limit' ],
+	expect: { '': {} }
+});
+
+
 function collectRemoteHosts (remotehosttableentries,Remotehosts) {
+	var ipAddrs = [];
 	for (var IPaddr in Remotehosts) {
-			remotehosttableentries.push([IPaddr, Remotehosts[IPaddr]['id']]);
-		}
+		if (typeof dns_cache[IPaddr] === 'undefined') ipAddrs.push(IPaddr);
+	}
+
+	L.resolveDefault(callNetworkRrdnsLookup(ipAddrs, 1000, 1000), {}).then(function(replies) {
+				for (var index in ipAddrs) {
+					var address = ipAddrs[index];
+					if (!address)
+						continue;
+					if (replies[address]) {
+						dns_cache[address] = replies[address];
+						continue;
+					} else {
+						for (var IPaddr in Remotehosts)
+							for (var mac in Hosts) {
+								for (var IPaddr2 in Hosts[mac]['ip6addrs'])
```

Comment:
Same for these two `for` loops I think

Reply:
Ok have a look. I replaced them with a .find(...)

Reply:
Looks great.

Reply:
Just a concern at this point: is it possible to enter that code block and the mac or Hosts is empty, producing a theoretical e.g.:

dns_cache[address]=Hosts[undefined]['name'];

You know what's happening best. Just checking.